h2o-automl/src/main/java/ai/h2o/automl/leaderboard/Leaderboard.java

```diff
@@ -575,43 +677,55 @@ public TwoDimTable toTwoDimTable(String... extensions) {
   }
 
   private TwoDimTable toTwoDimTable(String tableHeader, boolean leftJustifyModelIds, String... extensions) {
-    final Key<Model>[] modelKeys = _model_keys.clone(); // leaderboard can be retrieved when AutoML is still running: freezing current models state.
+    final Lock readLock = lock.readLock();
+    if (readLock.tryLock()) {
+      try {
+        final Key<Model>[] modelKeys = _model_keys.clone(); // leaderboard can be retrieved when AutoML is still running: freezing current models state.
+        final List<LeaderboardColumn> columns = getDefaultTableColumns();
+        final List<LeaderboardColumn> extColumns = new ArrayList<>();
+        if (getModelCount() > 0) {
+          final Key<Model> leader = getModelKeys()[0];
+          LeaderboardCell[] extCells = (extensions.length > 0 && LeaderboardExtensionsProvider.ALL.equalsIgnoreCase(extensions[0]))
+                  ? getExtensions(leader)
+                  : Stream.of(extensions).map(e -> getExtension(leader, e)).toArray(LeaderboardCell[]::new);
+          Stream.of(extCells).filter(Objects::nonNull).forEach(e -> extColumns.add(e.getColumn()));
+        }
+        columns.addAll(extColumns);
+
+        TwoDimTable table = makeTwoDimTable(tableHeader, modelKeys.length, columns.toArray(new LeaderboardColumn[0]));
+
+        int maxModelIdLen = Stream.of(modelKeys).mapToInt(k -> k.toString().length()).max().orElse(0);
+        final String[] modelIDsFormatted = new String[modelKeys.length];
+        for (int i = 0; i < modelKeys.length; i++) {
+          Key<Model> key = modelKeys[i];
+          if (leftJustifyModelIds) {
+            // %-s doesn't work in TwoDimTable.toString(), so fake it here:
+            modelIDsFormatted[i] = org.apache.commons.lang.StringUtils.rightPad(key.toString(), maxModelIdLen);
+          } else {
+            modelIDsFormatted[i] = key.toString();
+          }
+          addTwoDimTableRow(table, i,
+                  modelIDsFormatted[i],
+                  getMetrics(),
+                  extColumns.stream().map(ext -> getExtension(key, ext.getName())).toArray(LeaderboardCell[]::new)
+          );
+        }
+        return table;
+      } finally {
+        readLock.unlock();
+      }
+    } else {
+      return makeTwoDimTable(tableHeader, 0, getDefaultTableColumns().toArray(new LeaderboardColumn[0]));
+    }
+  }
```

Comment:
mainly reindentation due to the introduction of the readLock as the client polling can occur when leaderboard is updated.